### PR TITLE
Post skills

### PR DESCRIPTION
**Pull request by Angham116**
### Related:
related to #48 

- query ```checkSkills``` for ensuring that is a new skill.
- query ```insertSkills``` for add new skill.
- route  ```/api/v1/skills```  with method post 
- integration test for the route with its query.

### Summary:
how to test this PR: 
 -  ```git fetch origin post-skills```
 -  ```git checkout post-skills```
 -  ```git pull origin  post-skills```
 - ```npm run dev```
 - on postman run this route  ``` `/api/v1/skills  ``` with method ` post ` 
 and body    :
```javascript
{
  "id": 1,
  "name": "JAVAFX"
}
```

### Checklist Criteria
- [x] **Branch is up to date with master**
- [x] **The project runs**
- - [x] You can download and install locally, and run the project locally
- - [x] All existing and new features run without bugs
- - [x] The build script runs without error
- [x] **Code is consistent**
- - [x] Linter is used, no error messages
- - [x] Code is consistent with agreed team style
- - [x] Naming of files is consistent with project (e.g. all files named with camelCase)
- [x] **No console logs**
- - [x] Console.log is only used for crucial system messages (e.g. which port server is on).
- - [x] Any console.logs used for debugging have been removed.

### CODE REVIEW

File: test/postSkill.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  supertest(router)
+    .post('/api/v1/skills')
+    .send(body)
+    .expect(200)
+    .end((err, result) => {
+      if (err) {
+        t.error(err);
+        t.end();
+      }
+      t.deepEqual(result.body.data, fields, 'Should contain the same content');
```
Comment (tomduggan85) [inline test/postSkill.js]: @Angham116 it might make life easier for you if you only check that result.body.data.name equals JAVAFX, and ignore the value of ID. Then if your dB fake data changes, causing the inserted id in this test to change, the test won’t fail. Ultimately, the name is what is important to test so I think that would be a safe change. 

Also, it would be awesome to add a test which ensures that the endpoint returns an error when you try to add a duplicate skill name. If you have time later in the project :)

Comment (Angham116) [reply test/postSkill.js]: @tomduggan85  I have thought about it yesterday & I will do it, thanx

File: server/controllers/skills/addSkills.js [hunk 2]
```diff
@@ -1,3 +1,16 @@
-module.exports = (req, res) => {
-  res.send(' add new skills ');
+const { checkSkills, insertSkills } = require('../../database/queries/skills/index');
+
+module.exports = (req, res, next) => {
+  const { name } = req.body;
+  checkSkills(name)
```
Comment (Fatmasiam) [inline server/controllers/skills/addSkills.js]:
```javascript
checkSkills(name)
    .then((result) => {
      if (result.rows.length !== 0) {
        return insertSkills(name)
            .then(response => res.send({
                    error: null,
                    data: response.rows[0],
              }))
          } 
          next({ code: 400, msg: 'skill already exist' });
    })
    .catch(err => next(err));
```

File: server/controllers/skills/addSkills.js [hunk 3]
```diff
@@ -1,3 +1,16 @@
-module.exports = (req, res) => {
-  res.send(' add new skills ');
+const { checkSkills, insertSkills } = require('../../database/queries/skills/index');
+
+module.exports = (req, res, next) => {
+  const { name } = req.body;
```
Comment (ashatat) [inline server/controllers/skills/addSkills.js]: Please validate the name to make sure the client gets the proper error msg

File: server/controllers/skills/addSkills.js (hunk 2 above)
Comment (Angham116) [reply server/controllers/skills/addSkills.js]: @Fatmasiam 
```javascript
   if (result.rows.length !== 0) 
```
 ???? 
 we need to check if there isnt ```skill.name``` same with ```name``` from body req.

File: test/postSkill.js (hunk 1 above)
Comment (tomduggan85) [reply test/postSkill.js]: @Angham116 thanks for doing that!
